invokeai/app/api/routers/models.py

```diff
@@ -71,22 +75,23 @@ async def list_models() -> ModelsList:
 )
 async def update_model(
     model_request: CreateModelRequest
-) -> CreateModelRequest:
-    #Adds a new model
+) -> CreateModelResponse:
+    """Adds a new model to the active model configuration file."""
     try:
         ApiDependencies.invoker.services.model_manager.add_model(
-            model_name=model_request["name"],
-            model_attributes=model_request["info"],
+            model_name=model_request.name,
+            model_attributes=model_request.info,
             clobber=True,
         )
-        # How does Ckpt support deprecation change the above?
```

Comment:
Your `CreateModelResponse` class has `name` and `info` fields, no `status` field. You'll need to provide those, or change the response structure. I'd also recommend you try to run each of these endpoints through the openapi doc (swagger ui) to see if they work =). The code looks okay, but I'm not sure if it'll work.